tests/tests/index_config.rs

```diff
@@ -730,3 +730,30 @@ fn delete_index_aborted_maintains_tantivy_files(mut conn: PgConnection) {
         index_dir
     );
 }
+
+#[rstest]
+fn custom_enum(mut conn: PgConnection) {
+    r#"
+    CREATE TYPE color AS ENUM ('red', 'green', 'blue');
+    CREATE TABLE paradedb.index_config(id INTEGER, description TEXT, color color);
+    INSERT INTO paradedb.index_config VALUES (1, 'Item 1', 'red'), (2, 'Item 2', 'green');
+    "#
+    .execute(&mut conn);
+
+    r#"
+    CALL paradedb.create_bm25(
+        index_name => 'index_config_index',
+        table_name => 'index_config',
+        schema_name => 'paradedb',
+        key_field => 'id',
+        text_fields => paradedb.field('description'),
+        numeric_fields => paradedb.field('color')
+    )
+    "#
+    .execute(&mut conn);
+
+    let rows: Vec<(i32, String)> =
+        "SELECT id, description FROM paradedb.index_config WHERE id @@@ paradedb.term('color', 'red'::color)".fetch(&mut conn);
+
+    assert_eq!(rows, vec![(1, "Item 1".into())]);
+}
```

Comment:
It might be worth having a test for `parse` behavior here. Our interface around `parse` seems like it's always teetering around change, so having the behavior of an edge case like enum search be codified is probably worthwhile.

Do you think this requires a documentation entry? We have these caveats with enum search that users probably should be aware of:
1. Needs to be written as numeric ordinal in `parse`.
2. Index might be off after `ALTER TYPE BEFORE/AFTER` ?